src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -214,6 +214,7 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
                 var errorHandled = false;
                 var numberOfDeliveryAttempts = 0;
 
+                contextBag.Set(message);
```

Comment:
The ContextBag created by the transport gets merged into the ContextBag managed by Core.